StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1504AllAccessorsMustBeSingleLineOrMultiLine.cs

```diff
@@ -96,35 +96,46 @@ private void HandlePropertyDeclaration(SyntaxNodeAnalysisContext context)
 
             if (property.AccessorList == null
                 || property.AccessorList.IsMissing
-                || property.AccessorList.Accessors.Count != 2)
+                || property.AccessorList.Accessors.Count < 2)
             {
                 return;
             }
 
-            if (!this.AreAllAccessorsInMultiLines(property.AccessorList.Accessors)
-               && !this.AreAllAccessorsInOneLine(property.AccessorList.Accessors))
+            var accessorsAnalyzeResult = this.AnalyzeAccessorsLineSpans(property.AccessorList.Accessors);
+            if (accessorsAnalyzeResult.SingleLineAccessors.Count == 1
+                && accessorsAnalyzeResult.MultiLineAccessors.Count == 1)
             {
-                context.ReportDiagnostic(Diagnostic.Create(Descriptor, property.AccessorList.Accessors[1].GetLocation()));
+                context.ReportDiagnostic(Diagnostic.Create(Descriptor, accessorsAnalyzeResult.SingleLineAccessors.Single().Keyword.GetLocation()));
             }
 
         }
 
-        private bool AreAllAccessorsInOneLine(IEnumerable<AccessorDeclarationSyntax> accessors)
+        private AccessorsAnalysisResult AnalyzeAccessorsLineSpans(IEnumerable<AccessorDeclarationSyntax> accessors)
         {
-            return accessors.All(a =>
+            var result = new AccessorsAnalysisResult();
+
+            foreach (var accessorDeclarationSyntax in accessors)
             {
-                var fileLinePositionSpan = a.GetLocation().GetLineSpan();
-                return fileLinePositionSpan.StartLinePosition.Line == fileLinePositionSpan.EndLinePosition.Line;
-            });
+                var fileLinePositionSpan = accessorDeclarationSyntax.GetLocation().GetLineSpan();
+
+                if (fileLinePositionSpan.StartLinePosition.Line == fileLinePositionSpan.EndLinePosition.Line)
+                {
+                    result.SingleLineAccessors.Add(accessorDeclarationSyntax);
+                }
+                else
+                {
+                    result.MultiLineAccessors.Add(accessorDeclarationSyntax);
+                }
+            }
+
+            return result;
         }
 
-        private bool AreAllAccessorsInMultiLines(IEnumerable<AccessorDeclarationSyntax> accessors)
+        private class AccessorsAnalysisResult
         {
-            return accessors.All(a =>
-            {
-                var fileLinePositionSpan = a.GetLocation().GetLineSpan();
-                return fileLinePositionSpan.StartLinePosition.Line < fileLinePositionSpan.EndLinePosition.Line;
-            });
+            public List<AccessorDeclarationSyntax> MultiLineAccessors { get; set; } = new List<AccessorDeclarationSyntax>();
```

Comment:
Yeah, I will remove them (and the one below). Unfortunatelly, need to go now, but I will fix it in a few hours :) Thanks !